en/02_Developer_Guides/17_CLI/02_Hybrid_Commands.md

```diff
@@ -0,0 +1,172 @@
+---
+title: PolyCommand
+summary: Code that can be run both from the command line and via HTTP requests
+icon: tasks-alt
+---
+
+# `PolyCommand` {#polycommand}
+
+Some code needs to be accessible both via HTTP requests over the web and via CLI in your terminal. Common examples of this are building the database and flushing cache.
+
+The [`PolyCommand`](api:SilverStripe\PolyExecution\PolyCommand) class provides a consistent API for getting input and providing output regardless of the context where the code is run. The API for this class is intentionally similar to the API for the `Command` class provided by `symfony/console`.
+
+> [!TIP]
+> See [the symfony/console input documentation](https://symfony.com/doc/current/console/input.html#using-command-options) for more specific details about using `InputInterface` and `InputOption`. Note that we use the [`getOptions()`](api:SilverStripe\PolyExecution\PolyCommand::getOptions()) method instead of `configure()` to return an array of `InputOption` objects. `PolyCommand` does not allow arguments.
+>
+> See [the symfony/console colouring documentation](https://symfony.com/doc/current/console/coloring.html) for information about styling output. Note that all output (including outputting for HTTP requests) as well as the command description and help info can use the `symfony/console` styling format.
+
+```php
+namespace App\Cli\Command;
+
+use SilverStripe\PolyExecution\PolyCommand;
+use SilverStripe\PolyExecution\PolyOutput;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+class MyPolyCommand extends PolyCommand
+{
+    protected static string $commandName = 'app:my-command';
```

Comment:
```suggestion
    protected static string $commandName = 'app:MyPolyCommand';
```
As mentioned [here](https://github.com/silverstripe/silverstripe-framework/issues/11341#issuecomment-2367027143) I think we should be encouraging the ClassName style naming rather than url style